python/paddle/fluid/tests/unittests/mkldnn/test_matmul_mkldnn_op.py

```diff
@@ -161,5 +161,231 @@ def set_attributes(self):
         self.attrs = {'force_fp32_output': True}
 
 
+@skip_check_grad_ci(reason="DNNL's MatMul doesn't implement grad kernel.")
+class TestMatMulOpReshapeTranspose(OpTest):
+    def generate_data(self):
+        self.x = np.random.random([2, 128, 768]).astype("float32").reshape(
+            [2, 128, 12, 64]).transpose([0, 2, 1, 3])
+        self.y = np.random.random([2, 128, 768]).astype("float32").reshape(
+            [2, 128, 12, 64]).transpose([0, 2, 1, 3])
+        self.out = np.matmul(self.x, self.y.transpose([0, 1, 3, 2]))
+        self.shape_x = []
+        self.axis_x = []
+        self.shape_y = []
+        self.axis_y = []
+
+    def setUp(self):
+        # Set max isa, otherwise fails on SKX and earlier
+        os.environ["DNNL_MAX_CPU_ISA"] = "AVX"
+        self.op_type = "matmul"
+        self._cpu_only = True
+        self.use_mkldnn = True
+        self.transpose_y = True
+        self.generate_data()
+
+        self.inputs = {'X': self.x, 'Y': self.y}
+        self.attrs = {
+            'use_mkldnn': self.use_mkldnn,
+            'transpose_Y': self.transpose_y
+        }
+        if len(self.axis_x) > 0:
+            self.attrs['axis_X'] = self.axis_x
+        if len(self.axis_y) > 0:
+            self.attrs['axis_Y'] = self.axis_y
+        if len(self.shape_x) > 0:
+            self.attrs['shape_X'] = self.shape_x
+        if len(self.shape_y) > 0:
+            self.attrs['shape_Y'] = self.shape_y
+
+        self.outputs = {'Out': self.out}
+
+    def test_check_output(self):
+        self.check_output()
+
+
+class TestMatMulOpReshapeTranspose4DXFloat(TestMatMulOpReshapeTranspose):
+    def generate_data(self):
+        self.x = np.random.random([2, 128, 768]).astype("float32")
+        self.y = np.random.random([2, 128, 768]).astype("float32").reshape(
+            [2, 128, 12, 64]).transpose([0, 2, 1, 3])
```

Comment:
Because self.x and self.y will be assigned as input of matmul mkldnn op, they should be simple as we only check if matmul mkldnn op working. Do you think self.x and self.y could be simplified? 
when we calculate self.out, we can do a lot of transpose and reshape etc.

Reply:
it is as I explained on the call i need to compare reshape+transpose+matmul with numpy matmul